xdsl/dialects/arm/assembly.py

```diff
@@ -9,3 +12,19 @@ class AssemblyInstructionArg(abc.ABC):
     @abc.abstractmethod
     def assembly_str(self) -> str:
         raise NotImplementedError()
+
+
+class reg(AssemblyInstructionArg):
+    """
+    A wrapper around SSAValue to be printed in assembly.
+    Only valid if the type of the value is a RegisterType.
+    """
+
+    value: SSAValue
+
+    def __init__(self, value: SSAValue) -> None:
+        self.value = value
+
+    def assembly_str(self) -> str:
+        assert isinstance(self.value.type, RegisterType)
+        return self.value.type.register_name.data
```

Comment:
also I'm not too clear on why we would have a concrete class like this instead of just an accessor method inside the register class